test_agent/rust/src/testagent.rs

```diff
@@ -0,0 +1,293 @@
+                            u_uuri,
+                            Arc::clone(&cloned_listener) as Arc<dyn UListener>,
+                        )
+                        .await
+                }
+
+                constants::UNREGISTER_LISTENER_COMMAND => {
+                    let cloned_listener = Arc::clone(&arc_self);
+                    let wu_uuri: WrapperUUri = serde_json::from_value(json_data_value).unwrap(); // convert json to UMessage
+                    let u_uuri = wu_uuri.0;
+                    action_str = constants::UNREGISTER_LISTENER_COMMAND;
+                    self.utransport
+                        .unregister_listener(
+                            u_uuri,
+                            Arc::clone(&cloned_listener) as Arc<dyn UListener>, /*&cloned_listener*/
+                        )
+                        .await
+                }
+
+                _ => Ok(()),
+            };
+
+            let mut status_dict: HashMap<String, _> = HashMap::new();
+
+            match status {
+                Ok(()) => {
+                    let status = UStatus::default();
+                    status_dict.insert(
+                        "message".to_string(),
+                        status.message.clone().unwrap_or_default(),
+                    );
+                    status_dict.insert("code".to_string(), 0.to_string());
+                }
+                Err(u_status) => {
+                    status_dict.insert(
+                        "message".to_string(),
+                        u_status.message.clone().unwrap_or_default(),
+                    );
+                    let enum_number = match u_status.get_code() {
+                        UCode::OK => 0,
+                        UCode::INTERNAL => 13,
+                        UCode::ABORTED => 10,
+                        UCode::ALREADY_EXISTS => 6,
+                        UCode::CANCELLED => 1,
+                        UCode::DATA_LOSS => 15,
+                        UCode::DEADLINE_EXCEEDED => 4,
+                        UCode::FAILED_PRECONDITION => 9,
+                        UCode::INVALID_ARGUMENT => 3,
+                        UCode::NOT_FOUND => 5,
+                        UCode::OUT_OF_RANGE => 11,
+                        UCode::PERMISSION_DENIED => 7,
+                        UCode::RESOURCE_EXHAUSTED => 8,
+                        UCode::UNAUTHENTICATED => 16,
+                        UCode::UNAVAILABLE => 14,
+                        UCode::UNIMPLEMENTED => 12,
+                        UCode::UNKNOWN => 2,
+                    };
+                    status_dict.insert("code".to_string(), enum_number.to_string());
+                }
+            }
+
+            let json_message = JsonResponseData {
+                action: action_str.to_owned(),
+                data: status_dict.clone(),
+                ue: "rust".to_owned(),
+                test_id: test_id.to_string(),
+            };
+
+            <SocketTestAgent as Clone>::clone(self)
```

Comment:
Again, IMHO due for the same refactoring I talked about up [here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1581427127).